hedera-node/src/main/java/com/hedera/services/ledger/TransactionalLedger.java

```diff
@@ -385,6 +385,28 @@ public ResponseCodeEnum validate(final K id, final LedgerCheck<A, P> ledgerCheck
 		}
 	}
 
+	public ResponseCodeEnum validateFungibleAllowance(final K id, final LedgerCheck<A, P> ledgerCheck) {
```

Comment:
Imo these `validateFungibleAllowance()` and `validateNftAllowance()` methods don't really make sense on the `TransactionalLedger` type---and they also lead to double duplication of `if (entitiesLedger == null) {...` pattern.

Can't we instead just update `BalanceChange.checkUsing()` to look at the type of change, and then do the appropriate validation?

Reply:
So, we would send `merkleAccount` to the BalanceChange object through `checkUsing()` ?? And these validations are just for allowances or adapt further ?

Reply:
> So, we would send merkleAccount to the BalanceChange object through checkUsing() ?? 
Yep! We can discuss tomorrow morning, it shouldn't actually be a large change I think. 🤔